### ISSUE TITLE
[Bug]: Rebuilding native modules fails with Node.js 17 and latest node-gyp / Electron 

### ISSUE THREAD

**Issue by inukshuk** [electron/electron#32114]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

16.0.3

### What operating system are you using?

Other Linux

### Operating System Version

ArchLinux

### What arch are you using?

x64

### Expected Behavior

Rebuilding native modules should work with Node.js v17.

### Actual Behavior

I've stumbled over an issue rebuilding native modules with node-gyp that is probably related to [the changes referred to in the Electron 16 blog-post](https://www.electronjs.org/blog#building-native-modules). I'm rebuilding three native modules with node-gyp on Linux, Windows and macOS for x64, ia32, and arm64. After updating both node-gyp and and Electron to the latest versions, this still works fine using Node.js v16. However, using Node.js v17 (I tried 17.1.0 and 17.2.0) the rebuild always fails with this node-gyp error:

```
npm ERR! gyp: name 'openssl_fips' is not defined while evaluating condition 'openssl_fips != ""' in binding.gyp while trying to load binding.gyp
```

**Comment by GentikSolm** [comment]
Same exact situation here, using Arch on x64, node v17.1.0. Im assuming this is happening from the same native / electron modules; I got here by trying to install hydrogen for atom.

**Comment by GentikSolm** [comment]
I can also confirm that this is fixed by switching to node 16, in my case 16.13.1.
